run/test/specs/disappearing_message_checks_android.spec.ts

```diff
@@ -0,0 +1,429 @@
+  await setDisappearingMessage(device1, [
+    "1o1",
+    "Disappear after send option",
+    "10 seconds",
+  ]);
+  await device1.navigateBack(platform);
+  // Send a link
+  await device1.inputText("accessibility id", "Message input box", testLink);
+  await device1.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Message sent status: Sent",
+    maxWait: 20000,
+  });
+  // Accept dialog for link preview
+  await device1.clickOnElement("Enable");
+  // No preview on first send
+  await device1.clickOnElement("Send message button");
+  // Send again for image
+  await device1.inputText("accessibility id", "Message input box", testLink);
+  await sleepFor(100);
+  await device1.clickOnElement("Send message button");
+  // Make sure image preview is available in device 2
+  await device2.waitForTextElementToBePresent({
+    strategy: "id",
+    selector: "network.loki.messenger:id/linkPreviewView",
+  });
+  // Wait for 10 seconds to disappear
+  await sleepFor(10000);
+  await device1.hasElementBeenDeletedNew({
+    strategy: "id",
+    selector: "network.loki.messenger:id/linkPreviewView",
+    maxWait: 1000,
+  });
+  await device2.hasElementBeenDeletedNew({
+    strategy: "id",
+    selector: "network.loki.messenger:id/linkPreviewView",
+    maxWait: 1000,
+  });
+  await closeApp(device1, device2);
+}
+
+async function disappearingCommunityInviteMessage(
+  platform: SupportedPlatformsType
+) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const communityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
+  const communityName = "Testing All The Things!";
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  await newContact(platform, device1, userA, device2, userB);
+  await setDisappearingMessage(device1, [
+    "1o1",
+    "Disappear after send option",
+    "10 seconds",
+  ]);
+  await device1.navigateBack(platform);
+  await device1.clickOnElement("New conversation button");
+  await device1.clickOnElement("Join community");
+  await device1.inputText("accessibility id", "Community input", communityLink);
+  await device1.clickOnElement("Join community button");
+  // Wait for community to load
+  await sleepFor(1000);
+
+  await device1.clickOnElement("More options");
+  await device1.clickOnElement("Add Members");
+  await device1.clickOnElementByText({
+    strategy: "accessibility id",
+    selector: "Contact",
+    text: userB.userName,
+  });
+  await device1.clickOnElement("Done");
+  // Check device 2 for invitation from user A
+  await device2.waitForTextElementToBePresent({
+    strategy: "accessibility id",
+    selector: "Community invitation",
+    text: communityName,
+  });
+  // Wait for 10 seconds for message to disappear
+  await sleepFor(10000);
```

Comment:
would be nice to be able to set a fake clock like we do for unit testing, but those running on real device it seems unlikely to be possible.